diff -Naur ipx-1.1/Makefile ipx-1.1.new/Makefile
--- ipx-1.1/Makefile	2002-03-10 11:57:50.000000000 +0000
+++ ipx-1.1.new/Makefile	2004-05-27 09:45:29.000000000 +0000
@@ -1,4 +1,6 @@
-CFLAGS = -O2 -Wall
+DESTDIR=
+
+CFLAGS = -Wall
 UTILS = ipx_configure ipx_interface ipx_internal_net ipx_route
 CC=gcc
 
@@ -10,17 +12,7 @@
 install: $(UTILS)
 	for i in $(UTILS); \
 	do \
-		install --strip $$i /sbin; \
-		install $$i.8 /usr/man/man8; \
+		install -D -m 0755 --strip $$i $(DESTDIR)/sbin/$$i; \
+		install -D -m 0644 $$i.8 $(DESTDIR)/usr/share/man/man8/$$i.8; \
 	done
-	install init.ipx /etc/rc.d/init.d/ipx
-	install -m 0644 config.ipx /etc/sysconfig/ipx
-	rm -f /etc/rc.d/rc2.d/S15ipx
-	ln -sf /etc/rc.d/init.d/ipx /etc/rc.d/rc2.d/S15ipx
-	rm -f /etc/rc.d/rc3.d/S15ipx
-	ln -sf /etc/rc.d/init.d/ipx /etc/rc.d/rc3.d/S15ipx
-	rm -f /etc/rc.d/rc5.d/S15ipx
-	ln -sf /etc/rc.d/init.d/ipx /etc/rc.d/rc5.d/S15ipx
-	rm -f /etc/rc.d/rc6.d/K55ipx
-	ln -sf /etc/rc.d/init.d/ipx /etc/rc.d/rc6.d/K55ipx
 
